io.github.lpub3d.LPub3D.json

```diff
@@ -0,0 +1,172 @@
+{
+  "id": "io.github.lpub3d.LPub3D",
+  "runtime": "org.kde.Platform",
+  "runtime-version": "5.15",
+  "sdk": "org.kde.Sdk",
+  "command": "lpub3d24",
+  "rename-icon": "lpub3d",
+  "rename-desktop-file": "lpub3d.desktop",
+  "rename-appdata-file": "io.github.lpub3d.appdata.xml",
+  "build-options": {
+    "env": {
+      "LP3D_3RD_PARTY_DIR": "/app/lpub3d_linux_3rdparty",
+      "LP3D_DIST_DIR_PATH": "/app/lpub3d_linux_3rdparty",
+      "LDRAWDIR": "/app/lpub3d_linux_3rdparty/ldraw",
+      "PLATFORM_CODE": "osl",
+      "TARGET_CPU": "$(uname -m)",
+      "PLATFORM_VER": "$(. /etc/os-release 2>/dev/null; [ -n \"$VERSION_ID\" ] && echo $VERSION_ID || true)"
+    }
+  },
+  "cleanup": [
+    "./lpub3d_linux_3rdparty"
+  ],
+  "finish-args": [
+    "--device=dri",
+    "--share=ipc",
+    "--socket=x11",
+    "--filesystem=home",
+    "--env=TMPDIR=/var/tmp",
+    "--env=QT_ENABLE_HIGHDPI_SCALING=1"
+  ],
+  "modules": [
+    {
+      "name": "part-archives",
+      "buildsystem": "simple",
+      "build-commands": [
+        "unzip -d ${LP3D_3RD_PARTY_DIR} -q /app/lpub3d_linux_3rdparty/complete.zip",
+        "if test -d ${LP3D_3RD_PARTY_DIR}/ldraw; then echo 'LDraw library extracted.'; fi"
+      ],
+      "sources": [
+        {
+          "type": "archive",
+          "dest-filename": "complete.zip",
+          "dest": "/app/lpub3d_linux_3rdparty",
```

Comment:
I was surprised this work. `dest` shouldn't be the install directory. You should do that in the `build-commands` or `post-install`.

This apply on the others too.
You probably want to make this `"type": "file"` and unzip them in the `build-comands`

Reply:
Is the problem the shared location with the renderers install command ? Can the archives be placed at a location in, but not at the root of `lpub3d_linux_3rdparty` ? The approach I used is also how I build on Travis, AppVeyor and my build from source scripts. The 3rdParty dir is used to 'stage' the renderers and hold the libraries. At LPub3D install, the renderers, libraries and their related content are sourced from there so `post-install` will not be possible without rewriting the LPub3D install routines.

The documentation states all type can use `dest`; moreover, here is an example https://github.com/flathub/io.github.maurycyliebner.enve/blob/master/io.github.maurycyliebner.enve.yaml 

Cheers,

Reply:
I've set the script to install each renderer in it's own folder and I move the installed content to the proper 'staging' location at `post-install`.

Cheers,

Reply:
/app is not for downloading "source" file.
`dest` is the destination directory.

Note that `type: archive` mean the archive will be extracted in that directory. But this is not the installation of assets.
For that the best is just `type: file` and then in the `post-install` or `build-commands` to unzip in the destination location.

In the example you link `dest` as actually used to checkout a git tree in a subdir of the main source tree.